aria/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+//! Pure Rust implementation of the [ARIA] block cipher ([RFC 5794]).
+//!
+//! # ⚠️ Security Warning: Hazmat!
+//!
+//! This crate implements only the low-level block cipher function, and is intended
+//! for use for implementing higher-level constructions *only*. It is NOT
+//! intended for direct use in applications.
+//!
+//! USE AT YOUR OWN RISK!
+//!
+//! # Examples
+//! ```
+//! use aria::cipher::generic_array::GenericArray;
+//! use aria::cipher::{Key, Block, BlockEncrypt, BlockDecrypt, KeyInit};
+//! use aria::Aria128;
+//!
+//! let key = GenericArray::from([0u8; 16]);
+//! let mut block = GenericArray::from([0u8; 16]);
+//! // Initialize cipher
+//! let cipher = Aria128::new(&key);
+//!
+//! let block_copy = block.clone();
+//! // Encrypt block in-place
+//! cipher.encrypt_block(&mut block);
+//! // And decrypt it back
+//! cipher.decrypt_block(&mut block);
+//! assert_eq!(block, block_copy);
+//! ```
+//!
+//! [ARIA]: https://en.wikipedia.org/wiki/ARIA_(cipher)
+//! [RFC 5794]: https://tools.ietf.org/html/rfc5794
+
+#![no_std]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/media/26acc39f/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/26acc39f/logo.svg"
+)]
+#![deny(unsafe_code)]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![warn(missing_docs, rust_2018_idioms)]
+
+mod consts;
+
+pub use cipher;
+
+use cipher::{
+    consts::{U16, U24, U32},
+    AlgorithmName, BlockCipher, Key, KeyInit, KeySizeUser,
+};
+use core::fmt;
+
+#[cfg(feature = "zeroize")]
+use cipher::zeroize::{Zeroize, ZeroizeOnDrop};
+
+use crate::consts::{C1, C2, C3, DIFFUSE_CONSTS, SB1, SB2, SB3, SB4};
+
+#[inline(always)]
+fn diffuse(x: [u8; 16]) -> u128 {
+    DIFFUSE_CONSTS
+        .iter()
+        .zip(x)
+        .map(|(a, b)| a * b as u128)
+        .fold(0, |a, v| a ^ v)
+}
+
+fn a(x128: u128) -> u128 {
+    let x = x128.to_be_bytes();
+    diffuse([
+        x[0], x[1], x[2], x[3], x[4], x[5], x[6], x[7], x[8], x[9], x[10], x[11], x[12], x[13],
+        x[14], x[15],
+    ])
```

Comment:
You can simply write it as `diffuse(x.to_be_bytes())`.

Reply:
done